### ISSUE TITLE
Native renderer doesn't call when Navigating back from any page to page which has custom control

### ISSUE THREAD

**Issue by KarthikRajaAKR** [xamarin/Xamarin.Forms#6428]
### Description
I have used navigation page. My Main page has custom entry, for which I have created native renderer. Initially renderer works fine. I navigate to next page using PushAsync, then returning back to the main page, here native renderer doesn't call for custom entry.

I have used this procedure in DataForm control for your convenience I reproduced the issue in a simple sample.  

Issue reproduce in Android and UWP platforms, but in my sample, I tried with UWP.

### Steps to Reproduce

1. Run the attached sample in UWP platform. Note the native button.
2. Place breakpoint in NativeRenderer OnElementChanged method.
3. Click NextPage. See New Page
4. Click PreviousPage.
5. Note the view loaded but the NativeRenderer OnElementChange doesn't call

You can say that OnElementChanged only hit when I change anything on that view, but while navigating back from another page I did some changes to my BindingContext value, Since method not getting execute the changes does not make to my form.

### Expected Behavior
OnElementChanged override method should execute while returning back from the Navigation page.
### Actual Behavior
OnElementChanged should execute while returning back from the Navigation page.

### Basic Information

- Version with issue: 3.1.0.697729
- IDE: Visual studio 2017
- Platform Target Frameworks: 
  - iOS:  Haven't tested 
  - Android: Android 7.1
  - UWP:  16299 
- Android Support Library Version:  NA
- Affected Devices: Windows 10 Laptop
Android devices(Moto G5, MI A1)

### Screenshots
NA

### Reproduction Link
[SampleProject.zip](https://github.com/xamarin/Xamarin.Forms/files/3261198/SampleProject.zip)

**Comment by yanshouwang** [comment]
I also encountered this, I override the viewrenderer<T1,T2> and when the base.Dispose(disoposing) method called, the iOS platform OnElementChanged method is called correctly but the Android platform exit itself directly, the question is that if the OnElementChanged method is not called, the followd statements will not be called when dispose the control, and the behavior is not uniform in every platforms:
`if (e.OldElement != null) `
`{`
`    // Unsubscribe from event handlers`
`}`

**Comment by KarthikRajaAKR** [comment]
Any update on this?

**Comment by rmarinho** [comment]
Why would you expect `OnElementChanged`  to be called?, the page is still on the NavigationStack so it wasn't t disposed or reused.

**Comment by KarthikRajaAKR** [comment]
Hi @rmarinho,

Thank you for your response.

As I mentioned in my query, my view has not changed but while navigating back from another page to my form page BindingContext value has changed since OnElementChanged was not called my changed BindingContext value doesn't make any difference to my form.

Regards.

**Comment by KarthikRajaAKR** [comment]
Hi Team,

I replicated the issue with Xamarin entry default view**(Not custom renderer).**
**Steps to Replicate issue**

I have used two pages having the same BindingContext (static class so single object will be created for both pages)

1. Run the sample 
[XamarinPicker.zip](https://github.com/xamarin/Xamarin.Forms/files/3320750/XamarinPicker.zip)

2. Enter text in entry field and goto next page using button
3. Note that the entered text in the previous page reflects on the current entry 
4. Edit the text in the entry. Note value updated to the corresponding text property
5. Move back to main page, note value not updated in mainpage's entry field. But if you edit the current text in main page, Name property will hold original value that should be in current entry field. Hence value updated but not reflected in MainPage since OnElementChanged not called.

I think I have explained it in clear.

Regards.

**Comment by samhouts** [comment]
Yeah, that'll be fun to fix. This is something that does work in plain UWP, so we should strive to make it work in Forms. Thanks!

**Comment by AjitKRuin** [comment]
Any update on this?

**Comment by carbonete** [comment]
Hi,
I'm wayting for this correction, any news ?

Thanks

**Comment by KarthikRajaAKR** [comment]
Hi @samhouts,

Could you please update on this? due to this, we are facing issues in our control. Hope it will be resolved in upcoming releases.

**Comment by Ganeshamoorthy10** [comment]
Hi @samhouts , any update on @KarthikRajaAKR's query?

**Comment by samhouts** [comment]
This is on the backlog. I don't currently have an estimate of when we may begin work on this item. Thank you for your patience!

**Comment by jayaleshwari** [comment]
@samhouts , Can you please update on this?

**Comment by samhouts** [comment]
This is still on the backlog. Thanks!

**Comment by carbonete** [comment]
> This is still on the backlog. Thanks!

Any news ? 
Thanks

**Comment by jayaleshwari** [comment]
Hi @samhouts ,

Any update on this.?

**Comment by Muniappan612** [comment]
Hi @samhouts ,

Any update on this.?

**Comment by jayaleshwari** [comment]
Hi @samhouts,

Any update on this. ?

**Comment by saiganeshls2504** [comment]
Hi @samhouts ,

Any update on this?

**Comment by Muniappan612** [comment]
Hi @samhouts,

Any update on this?

**Comment by saiganeshls2504** [comment]
Hi @samhouts ,

Any update on this?

**Comment by Muniappan612** [comment]
Hi @samhouts ,

Any update on this?

**Comment by saiganeshls2504** [comment]
Hi @samhouts ,

Can you please any update for this issue?

**Comment by saiganeshls2504** [comment]
any update??

**Comment by saiganesh-sakthivel** [comment]
Hi @samhouts ,

Can you please tell any updates for this issue?
